htdocs/takepos/ajax/ajax.php

```diff
@@ -151,6 +151,7 @@
 					$sql .= " AND EXISTS (SELECT cp.fk_product FROM " . $db->prefix() . "categorie_product as cp WHERE cp.fk_product = p.rowid AND cp.fk_categorie IN (".$db->sanitize($filteroncategids)."))";
 				}
 				$sql .= " AND tosell = 1";
+				$sql .= " AND (barcode IS NULL OR barcode != '" . $db->escape($term) . "')";
```

Comment:
If the product PR1 has a barcode BBB, and with search on "PR", the result will return an empty string when we want to get the product ?
This was the goal of the line
$sql .= " AND ref = '" . $db->escape($barcode_value_list['ref']) . "'";
If you add this new line, the search will always fails when searching on a barcode_rules_list that requests a search on a ref.
When doing a 
if (isset($barcode_value_list['ref'])) {
my understanding is that we want a search on ref whatever is barcode.

Reply:
@eldy 
No, this part of code it's only used when you have a barcorde rule to insert product with a quantity (such as product whose barcode depends on weight)
So you can't set a barcode in this kind of product or have only one barcode for only one unit of product.
And the search is only done when the number of characters in length of term matches with the number of characters in barcode rule "($barcode_length == $barcode_char_nb)"

I kept the old method of search (see at bottom of this code).